shared/health/src/main/java/io/pravega/shared/health/impl/HealthEndpointImpl.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright Pravega Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pravega.shared.health.impl;
+
+import io.pravega.shared.health.ContributorNotFoundException;
+import io.pravega.shared.health.ContributorRegistry;
+import io.pravega.shared.health.Health;
+import io.pravega.shared.health.HealthContributor;
+import io.pravega.shared.health.HealthEndpoint;
+import io.pravega.shared.health.Status;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+@Slf4j
+public class HealthEndpointImpl implements HealthEndpoint {
+
+    private final ContributorRegistry registry;
+
+    /**
+     * Creates a new instance of the {@link HealthEndpointImpl} class.
+     * @param registry The {@link ContributorRegistry} to query.
+     */
+    HealthEndpointImpl(ContributorRegistry registry) {
+        this.registry = registry;
+    }
+
+    /**
+     * Provides the name of the root {@link HealthContributor}.
+     * @return The name (its id).
+     */
+    @Override
+    public String getDefaultContributorName() {
+        return registry.getRootContributor().getName();
+    }
+
+    /**
+     * Validates that the {@link HealthContributor} exists and requests it's {@link Health}.
+     *
+     * @param id  The id/name of the {@link HealthComponent} to check the {@link Health} of.
+     * @param includeDetails Whether or not to include detailed information provided by said {@link HealthComponent}.
+     * @return The {@link Health} result of the {@link HealthContributor} with name 'id'.
+     */
+    @NonNull
+    @Override
+    public Health getHealth(String id, boolean includeDetails) {
+        HealthContributor result = registry.get(id);
+        if (Objects.isNull(result)) {
```

Comment:
== null

But why not just return null??? You can define in the contract that "if not registered, it will return null".